### PR TITLE
Optimist state sync pagination

### LINKED ISSUE #9: Pagination on optimist state-sync getBlocks function #67

### PR DESCRIPTION
**Pull request by pawelgrzybek**
## What does this implement/fix? Explain your changes.

When optimist sync is triggered, it is going to chunk blocks to avoid replaying all events at once.

## Does this close any currently open issues?

fix #9 

## What commands can I run to test the change?

```
cd common-files && NODE_CONFIG_DIR=../config mocha --timeout 0 --bail --exit test/calculate-block-chunks.test.mjs
```

## Any other comments?

If you prefer to use an other value than 1000 as a default block chunk, I am happy to change it. For now it shouldn't make any difference until it is not prod deploy.

### CODE REVIEW

**Review by druiz0992 — CHANGES_REQUESTED** [review]
`CONFIRMATIONS` env variable should not be mixed with `STATE_SYNC_BLOCKS_CHUNK`

**Comment by druiz0992** [comment]
Sync test is failing. This is the error reported:
```optimist_1                 | [2023-03-20 15:47:56.096] ERROR: pastShieldEvents is not iterable
optimist_1                 |     err: {
optimist_1                 |       "type": "TypeError",
optimist_1                 |       "message": "pastShieldEvents is not iterable",
optimist_1                 |       "stack":
optimist_1                 |           TypeError: pastShieldEvents is not iterable
optimist_1                 |               at syncState (file:///app/src/services/state-sync.mjs:87:10)
optimist_1                 |               at processTicksAndRejections (node:internal/process/task_queues:96:5)
optimist_1                 |               at async initialBlockSync (file:///app/src/services/state-sync.mjs:176:7)
optimist_1                 |               at async main (file:///app/src/index.mjs:60:5)
optimist_1                 |     }
```

**Review by daveroga — APPROVED** [review]

File: nightfall-optimist/src/routes/block.mjs [hunk 1]
```diff
@@ -28,7 +28,7 @@ router.post('/check', async (req, res, next) => {
 
 router.post('/make-now', async (req, res, next) => {
   try {
-    logger.debug(`block make-now endpoint received GET`);
+    logger.debug(`block make-now endpoint received POST`);
```
Comment (imagobea) [inline nightfall-optimist/src/routes/block.mjs:31]: The log shouldn't be needed at all since `httpUtils` logs some info for all requests/responses

Comment (pawelgrzybek) [reply nightfall-optimist/src/routes/block.mjs:31]: For the sake of correctness. We can open a task and remove individual logs like that in one go, but that would be out of scope of this issue.

**Review by druiz0992 — APPROVED** [review]

**Pull request merged**